addons/calendar/models/calendar_event.py

```diff
@@ -492,11 +492,7 @@ def _compute_unavailable_partner_ids(self):
                 start, stop)
             for event in events:
                 for partner in event.partner_ids:
-                    if any(
-                        intervals_overlap(
-                            (event.start, event.stop), (other_event.start, other_event.stop))
-                        for other_event in events_by_partner_id.get(partner._origin.id, []) if other_event != event
-                    ):
+                    if event._is_partner_unavailable(partner, events_by_partner_id.get(partner._origin.id, [])):
```

Comment:
`events_by_partner_id.get(partner._origin.id, [])` will crash when we do a filtered later in appointment, so instead of using an empty list when we found nothing we should use `self.env["calendar.event"]`. Like this we make sure we always use the same type of info for that argument.